Tests/Unit/MiniAppDownloaderTests.swift

```diff
@@ -0,0 +1,91 @@
+import Quick
+import Nimble
+@testable import MiniApp
+
+// swiftlint:disable function_body_length
+class MiniAppDownloaderTests: QuickSpec {
+
+    override func spec() {
+        describe("mini app folder will be created") {
+            context("when valid manifest information is returned") {
+                it("will be downloaded and path is returned") {
+                    let mockAPIClient = MockAPIClient()
+                    let downloader = MiniAppDownloader()
+                    let responseString = """
+                      {
+                        "id": "123",
+                        "versionTag": "1",
+                        "name": "Test",
+                        "files": ["https://devstreaming-cdn.apple.com/videos/tutorials/TestFlight_App_Store_Connect_2018/TestFlight_App_Store_Connect_2018_sd.mp4",
+                        ]
+                      }
+                    """
+                    mockAPIClient.data = responseString.data(using: .utf8)
+                    var miniApp: URL?
+                    downloader.download(with: mockAPIClient, appId: "Apple", versionId: "Mac") { (result) in
+                        switch result {
+                        case .success(let url):
+                            miniApp = url
+                        case .failure:
+                            break
+                        }
+                    }
+                    let miniAppPath = FileManager.getMiniAppDirectory(with: "Apple", and: "Mac")
+                    expect(miniApp?.path).toEventually(equal(miniAppPath?.path), timeout: 50)
+                }
+            }
+        }
+        describe("mini app files will be downloaded") {
+            context("when valid manifest information is returned ") {
+                it("will be downloaded and path is returned") {
+                    let mockAPIClient = MockAPIClient()
+                    let downloader = MiniAppDownloader()
+                    let responseString = """
+                      {
+                        "id": "123",
+                        "versionTag": "1",
+                        "name": "Test",
+                        "files": ["https://devstreaming-cdn.apple.com/videos/tutorials/TestFlight_App_Store_Connect_2018/TestFlight_App_Store_Connect_2018_sd.mp4",
```

Comment:
I think that `MiniAppClient#download` isn't mocked, right? So this is downloading the real file? 

Can you add a mock for `download` and have it return a mock file or some static file from the repo?

Reply:
Updated. Mocked the download progress now by.